helpers/DeliveryHelperService.php

```diff
@@ -0,0 +1,287 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoProctoring\helpers;
+
+use common_ext_ExtensionsManager;
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\service\ConfigurableService;
+use oat\oatbox\user\User;
+use oat\taoDelivery\model\execution\DeliveryExecutionInterface;
+use oat\taoProctoring\model\execution\DeliveryExecution;
+use oat\taoProctoring\model\monitorCache\DeliveryMonitoringService;
+use oat\taoProctoring\model\TestSessionConnectivityStatusService;
+use oat\taoQtiTest\models\SessionStateService;
+
+/**
+ * Class DeliveryHelperService
+ * @author Bartlomiej Marszal
+ */
+class DeliveryHelperService extends ConfigurableService
```

Comment:
So, based on what we discussed in person, here are a few suggestions how to apply the boy scout rule (https://medium.com/@biratkirat/step-8-the-boy-scout-rule-robert-c-martin-uncle-bob-9ac839778385) on this task :)

- Use Cover and modify approach before refactoring (https://docs.google.com/presentation/d/1lnmxPT50g8C8yeAnE_lnQWjJDshO1o6jWWk6Kc6T1zM/edit#slide=id.g596cf22888_0_96)  
- Try to get rid of the ServiceManager, inject the actual dependencies via constructor
- Try to wrap static helper classes with other objects and inject them as in constructor
- Reduce Npath and cyclomatic complexity of the methods (https://modess.io/npath-complexity-cyclomatic-complexity-explained/)
-  Try to enforce the object calisthenics rules (one level indentation per method, avoid ELSE clauses, etc... https://williamdurand.fr/2013/06/03/object-calisthenics/)
- Try to apply single responsibility principle, this class does way too many things
- Get rid of the inconsistent return points of your methods (true|false|null, etc)
- Try to introduce value objects instead of working with plain arrays

I think this is enough for now :)

Reply:
I apply most of your points into my changes. Some of them were to time consuming and I couldn't perform them because it's only bug fix

Reply:
OK I'll approve, there is no need to push this further in scope of this bugfix. Nice job 👍